### ISSUE TITLE
Firefox crashes

### ISSUE THREAD

**Issue by NoSavvy** [KaOSx/apps#233]
#### Submission type

  - [ ] Bug report
  - [ ] Feature Request

#### Package name & version you are having issues with or have a request for

> …Firefox ver. 132-1 & 131-3

#### Used repositories

> …Base Kaos Repos

#### Expected behaviour

> …Not to crash

#### Actual behaviour

> …Crash after second opening

#### Steps to reproduce the problem

> …A fresh september ISO installed and updated.
Rebooted and installed FF and en-gb language.
Open FF from the command line (ok) load 3 sites, (BBC,Amazon and Vodafone) and pin them.
Exit FF and no errors reported to the CL, but about 20-30 secs for the prompt to reappear.
Open FF again from the CL (ok) click on a tab and FF crashes.

#### System info link created by Octopi > Tools > SysInfo

> …https://pastebin.com/UBjKQxuH

**Comment by demmm** [comment]
Since this is not running in a Wayland session, please test if running Firefox in non-wayland mode makes any difference:
```
MOZ_ENABLE_WAYLAND=0 firefox
```

**Comment by demmm** [comment]
Can you post as much as possible of the cli output when the crash occurs? Or is there nothing more then the repetition of `Exiting due to channel error.`?
I have found many, many threads and bug reports that show those lines, but not enough to pinpoint any recent cause (only one vaguely showing some related is nss post 3.104)

**Comment by NoSavvy** [comment]
Crash info this morning:-
Nov 02 09:48:32 john-Blackbox dbus-daemon[406]: [system] Activating via systemd: service name='org.freedesktop.timedate1' unit='dbus-org.freedesktop.timedate1.service' requested by ':1.334' (uid=1000 pid=8309 comm="firefox")
Nov 02 09:48:55 john-Blackbox systemd-coredump[8815]: Process 8309 (firefox) of user 1000 dumped core.
Nov 02 09:48:55 john-Blackbox drkonqi-coredump-processor[8816]: "/usr/lib/firefox/firefox" 8309 "/var/lib/systemd/coredump/core.firefox.1000.5fdcaad629df468aa6607bdd4f3e7474.8309.1730540932000000.zst"
Nov 02 09:48:55 john-Blackbox drkonqi-coredump-launcher[8832]: Unable to find file for pid 8309 expected at "kcrash-metadata/firefox.5fdcaad629df468aa6607bdd4f3e7474.8309.ini"
Nov 02 09:50:30 john-Blackbox dbus-daemon[406]: [system] Activating via systemd: service name='org.freedesktop.timedate1' unit='dbus-org.freedesktop.timedate1.service' requested by ':1.351' (uid=1000 pid=8883 comm="firefox")
Nov 02 09:51:04 john-Blackbox systemd-coredump[9452]: Process 8883 (firefox) of user 1000 dumped core.
Nov 02 09:51:04 john-Blackbox drkonqi-coredump-processor[9453]: "/usr/lib/firefox/firefox" 8883 "/var/lib/systemd/coredump/core.firefox.1000.5fdcaad629df468aa6607bdd4f3e7474.8883.1730541060000000.zst"
Nov 02 09:51:04 john-Blackbox drkonqi-coredump-launcher[9475]: Unable to find file for pid 8883 expected at "kcrash-metadata/firefox.5fdcaad629df468aa6607bdd4f3e7474.8883.ini"

Crash file from /var/lib/systemd/coredump is too big at 47MB

**Comment by NoSavvy** [comment]
I have tried all methods and no useful output.
The following is output when first starting with no .mozilla directory :-

Crash Annotation GraphicsCriticalError: |[0][GFX1-]: GFX: CanvasTranslator failed creating WebGL shared context (t=26.6378) [GFX1-]: GFX: CanvasTranslator failed creating WebGL shared context 

This is on different PC's with different graphics.

**Comment by f35f0ef9d0e827dae86552d3899f78fc** [comment]
I recently put KaOS on my laptop after playing with it in a VM for probably a couple years (along with other distros).

I am experiencing very frequent crashes in Firefox as well, but they do not occur on my other distributions, nor do they occur when I install from Flatpak in KaOS. When I find a website that causes the crash, I can reproduce it by browsing to the same site. I tried using a new Firefox profile (no extensions) as well. No difference.

I am using **Firefox 132.0.1**

The only output I get from the command line is here:

>[Parent 36051, IPC I/O Parent] WARNING: process 36286 exited on signal 11: file /buildsys/apps/firefox/src/firefox-132.0.1/ipc/chromium/src/base/process_util_[posix.cc:335](http://posix.cc:335/)

Here is one website sample that causes the crash:
https://www.missallieskitchen.com/venison-backstrap-recipe/

**Comment by demmm** [comment]
Recently, I have the same issue, it is even that far now, Firefox won't stay up at all.
Again, so far nothing in the logs that point to any possible fix.
No issue at all to use one of the Firefox provided binaries, just downloaded & ran from the download location, so clearly it is a KaOS packaging issue.
Not clear though if it is related to newer Firefox, or if it is because of an update to one of its' depends (nss?).

It might be needed to temporary provide a Firefox-bin instead of compiled, since an unusable Firefox can't continue.

**Comment by demmm** [comment]
After several rebuilds (using older and ESR versions) with no difference, Firefox has moved to a bin version for now.
Up in the build repo at this time, please test & report back if this version works again (for those not using the build repo):
```
sudo pacman -U http://kaosx.tk/repo/build/firefox-132.0.2-2-x86_64.pkg.tar.zst
```

**Comment by f35f0ef9d0e827dae86552d3899f78fc** [comment]
> After several rebuilds (using older and ESR versions) with no difference, Firefox has moved to a bin version for now. Up in the build repo at this time, please test & report back if this version works again (for those not using the build repo):
> 
> ```
> sudo pacman -U http://kaosx.tk/repo/build/firefox-132.0.2-2-x86_64.pkg.tar.zst
> ```

Yes, this solved the problem for me. At first, it started with a blank profile, but then I launched with `firefox -P` to select my old profile, and I got all my settings back and the problem was still fixed.
